docs/source/development/automated_testing.md

```diff
@@ -0,0 +1,165 @@
+│           │   ...
+│           │   nodes.py
+│           │   ...
+│   
+└───tests
+│   └───pipelines
+│       └───dataprocessing
+│           │   ...
+│           │   test_nodes.py
+│           │   ...
+```
+
+### Create an example test
+
+Now that you have a place to put your tests, you can create an example test in the new file `/src/tests/test_run.py`. The example test simply checks that the project_path attribute of a specially-defined `KedroContext` object has been correctly set.
+
+```
+import pytest
+from kedro.config import ConfigLoader
+from kedro.framework.context import KedroContext
+from kedro.framework.hooks import _create_hook_manager
+
+
+@pytest.fixture
+def config_loader():
+    return ConfigLoader(conf_source=str(Path.cwd()))
+
+
+@pytest.fixture
+def project_context(config_loader):
+    return KedroContext(
+        package_name=<package_name>,
+        project_path=Path.cwd(),
+        config_loader=config_loader,
+        hook_manager=_create_hook_manager(),
+    )
+
+class TestProjectContext:
+    def test_project_path(self, project_context):
+        assert project_context.project_path == Path.cwd()
+```
+
+This test is redundant, but it introduces a few of `pytest`'s core features and demonstrates the layout of a test file:
+- [Fixtures](https://docs.pytest.org/en/7.1.x/explanation/fixtures.html#about-fixtures) are used to define resources used in tests.
+- Tests are implemented in methods or functions beginning with `test_` and classes beginning with `Test`.
+- The `assert` statement is used to compare the result of the test with an expected value.
+
+You can read more about the basics of using `pytest` [here](https://docs.pytest.org/en/7.1.x/getting-started.html). For help writing your own tests and using all of the features of `pytest`, see the [docs](https://docs.pytest.org/).
+
+### Run your tests
+
+To run your tests, run `pytest` from within your project's root directory or from the `/src` directory.
+
+```bash
+cd <project_root>
+pytest
+```
+
+If you created the example test in the previous section, you should see the following output in your shell.
+
+```
+============================= test session starts ==============================
+...
+collected 1 item
+
+src/tests/test_run.py .                                                  [100%]
+
+============================== 1 passed in 0.38s ===============================
+```
+
+This output indicates that one test ran successfully in the file `src/tests/test_run.py`.
+
+## Add test coverage reports with `pytest-cov`
+
+It can be useful to see how much of your project is covered by tests. For this, you can install and configure the [`pytest-cov`](https://pypi.org/project/pytest-cov/) plugin for `pytest`, which is based on the popular [`coverage.py`](https://coverage.readthedocs.io/) library.
+
+### Install `pytest-cov`
+
+Install `pytest` as you would install other packages with pip, making sure your project's virtual environment is active.
```

Comment:
Could link to the virtual env setup docs again here.

Reply:
Added in [13586b6](https://github.com/kedro-org/kedro/pull/1879/commits/13586b61d8d9c81d5196e0358ee010ccb4e710e7).